notebooks/community/vizier/conversions_vertex_vizier_and_open_source_vizier.ipynb

```diff
@@ -0,0 +1,624 @@
+        "    from google.cloud.aiplatform.vizier import Study, pyvizier\n",
+        "\n",
+        "    def CreateStudy(project, location, display_name, problem):\n",
+        "        aiplatform.init(project=project, location=location)\n",
+        "        study = Study.create_or_load(display_name=display_name, problem=problem)\n",
+        "        return study\n",
+        "\n",
+        "else:\n",
+        "    from vizier.service import clients, pyvizier\n",
+        "\n",
+        "    def CreateStudy(project, location, display_name, problem):\n",
+        "        clients.environment_variables.service_endpoint = SERVICE_ENDPOINT\n",
+        "        study = clients.Study.from_study_config(\n",
+        "            problem, owner=OWNER, study_id=STUDY_DISPLAY_NAME\n",
+        "        )\n",
+        "        return study"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "guvkcQe_-zQf"
+      },
+      "source": [
+        "### Metric evaluation functions\n",
+        "\n",
+        "Next, define some functions to evaluate the two objective metrics.\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "Fjfk5_c900Oz"
+      },
+      "outputs": [],
+      "source": [
+        "# r * sin(theta)\n",
+        "def Metric1Evaluation(r, theta):\n",
+        "    \"\"\"Evaluate the first metric on the trial.\"\"\"\n",
+        "    return r * math.sin(theta)\n",
+        "\n",
+        "\n",
+        "# r * cos(theta)\n",
+        "def Metric2Evaluation(r, theta):\n",
+        "    \"\"\"Evaluate the second metric on the trial.\"\"\"\n",
+        "    return r * math.cos(theta)\n",
+        "\n",
+        "\n",
+        "def CreateMetrics(r, theta):\n",
+        "    # Evaluate both objective metrics for this trial\n",
+        "    y1 = Metric1Evaluation(r, theta)\n",
+        "    y2 = Metric2Evaluation(r, theta)\n",
+        "    print(\n",
+        "        \"[r = {}, theta = {}] => y1 = r*sin(theta) = {}, y2 = r*cos(theta) = {}\".format(\n",
+        "            r, theta, y1, y2\n",
+        "        )\n",
+        "    )\n",
+        "    measurement = pyvizier.Measurement()\n",
+        "    measurement.metrics[\"y1\"] = y1\n",
+        "    measurement.metrics[\"y2\"] = y2\n",
+        "\n",
+        "    # Return the results for this trial\n",
+        "    return measurement"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "2DgUIEpZ-_fJ"
+      },
+      "source": [
+        "### Optimization\n",
+        "\n",
+        "The following code defines a study with parameters and metrics. Then evaluates the metric information based on the suggestions from Vizier and reports the metrics value back. After a few round of iteration, you can get optimal trials by calling optimal_trials(). The code is adapt on both Vertex Vizier and OSS Vizier."
```

Comment:
"The following code defines a study with parameters and metrics, evaluates the metric information based on the suggestions from Vizier, and reports the metrics value back. After a few rounds of iteration, you can get optimal trials by calling `optimal_trials()`. The code is adapt to both Vertex Vizier and OSS Vizier."